omod/src/main/java/org/openmrs/module/appui/rest/SessionController.java

```diff
@@ -47,6 +47,16 @@ public Object get(HttpServletRequest request, HttpServletResponse response) {
         return ret;
     }
 
+    @RequestMapping(method = RequestMethod.POST)
+    @ResponseBody
+    public Object setLocation(@RequestParam("location") Location location,
```

Comment:
I think this needs to be a proper RESTful endpoint, i.e. one posts a @RequestBody of json/xml on an object that contains a "sessionLocation" property like

    { uuid: "uuid-of-location"}